grid.v3/codec/registry.go

```diff
@@ -1,61 +1,90 @@
 package codec
 
 import (
+	"errors"
 	"reflect"
 	"sync"
+
+	"github.com/gogo/protobuf/proto"
 )
 
-var registry *codecRegistry = &codecRegistry{reg: map[string]Codec{}} //shared registry
+var (
+	ErrNonProtoMessage         = errors.New("codec: non proto message")
```

Comment:
I think we should call this `ErrNonSupportedMessage` or something.    Because nothing in this PR excludes using some other messaging type.   For example with minor changes to Register, Un/Marshal we could support something else.